test/auth.spec.js

```diff
@@ -0,0 +1,65 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require('../server/index');
+const knex = require('../dbConn');
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+let register;
+
+describe('AUTH API ENDPOINTS', () => {
+  before(async () => {
+    await knex.migrate.latest();
+  });
+  beforeEach(() => {
+    register = {
+      firstName: 'john',
+      lastName: 'doe',
+      email: 'john@gmail.com',
+      password_confirmation: 'qwer1234',
+      password: 'qwer1234'
+    };
+  });
+  after(async () => {
+    await knex('users').truncate();
+    knex.destroy();
+    app.close();
+  });
+
+  describe('POST SIGN UP api/v1/auth/signup', () => {
+    it('should not sign up user if email is empty', async () => {
+      try {
+        register.email = '';
+        const res = await chai.request(app)
+          .post('/api/v1/auth/signup').send(register);
+        expect(res).to.have.status(400);
+        expect(res.body).to.be.an('array');
```

Comment:
When writing assertions, use this as a guide. If Anything changes in your app and your tests do not fail then your assertions are not the best yet. For example, to make this test pass, all I need to do is respond with 

```js
res.status(400).json([{ message: null }])
```

And your test passes. This is indication that your test does not actually test how the application works.